amd_tressfx_vulkan/src/UtilVulkan.cpp

```diff
@@ -0,0 +1,375 @@
+    pipelineCreateInfo.basePipelineIndex = -1;
+    pipelineCreateInfo.pRasterizationState = &rasterizationCreateInfo;
+    pipelineCreateInfo.pInputAssemblyState = input_assembly;
+    pipelineCreateInfo.pStages = stages;
+    pipelineCreateInfo.layout = layout;
+    pipelineCreateInfo.renderPass = render_pass;
+    pipelineCreateInfo.subpass = subpass;
+    return pipelineCreateInfo;
+}
+
+namespace
+{
+VkPipelineDepthStencilStateCreateInfo
+getSpecializedDSS(bool depth_test_enable, bool depth_write_enable,
+                  bool stencil_test_enable, VkCompareOp stencil_op, VkStencilOp pass_op,
+                  uint32_t write_mask)
+{
+    VkPipelineDepthStencilStateCreateInfo DSS{
+        VK_STRUCTURE_TYPE_PIPELINE_DEPTH_STENCIL_STATE_CREATE_INFO};
+    DSS.depthTestEnable = depth_test_enable;
+    DSS.depthCompareOp = VK_COMPARE_OP_LESS_OR_EQUAL;
+    DSS.depthWriteEnable = depth_write_enable;
+    DSS.stencilTestEnable = stencil_test_enable;
+    DSS.front.compareMask = 0xff;
+    DSS.front.writeMask = write_mask;
+    DSS.front.failOp = VK_STENCIL_OP_KEEP;
+    DSS.front.depthFailOp = VK_STENCIL_OP_KEEP;
+    DSS.front.passOp = pass_op;
+    DSS.front.compareOp = stencil_op;
+    DSS.back.compareMask = 0xff;
+    DSS.back.writeMask = write_mask;
+    DSS.back.failOp = VK_STENCIL_OP_KEEP;
+    DSS.back.depthFailOp = VK_STENCIL_OP_KEEP;
+    DSS.back.passOp = pass_op;
+    DSS.back.compareOp = stencil_op;
+    return DSS;
+}
+
+const VkPipelineColorBlendAttachmentState BlendStateBlendToBg_render_target_0{
+    true, VK_BLEND_FACTOR_ONE, VK_BLEND_FACTOR_SRC_ALPHA, VK_BLEND_OP_ADD,
+    VK_BLEND_FACTOR_ZERO, VK_BLEND_FACTOR_ZERO, VK_BLEND_OP_ADD,
+    VK_COLOR_COMPONENT_R_BIT | VK_COLOR_COMPONENT_A_BIT | VK_COLOR_COMPONENT_B_BIT |
+        VK_COLOR_COMPONENT_G_BIT};
+
+const VkPipelineColorBlendAttachmentState m_pDepthWritesToColor_BS_render_target_0{
+    true,
+    VK_BLEND_FACTOR_ZERO,
+    VK_BLEND_FACTOR_SRC_COLOR,
+    VK_BLEND_OP_ADD,
+    VK_BLEND_FACTOR_ZERO,
+    VK_BLEND_FACTOR_SRC_ALPHA,
+    VK_BLEND_OP_ADD,
+    VK_COLOR_COMPONENT_R_BIT};
+
+const VkPipelineColorBlendAttachmentState m_pResolveColor_BS_BS_render_target_0{
+    true, VK_BLEND_FACTOR_ONE, VK_BLEND_FACTOR_SRC_ALPHA, VK_BLEND_OP_ADD,
+    VK_BLEND_FACTOR_ZERO, VK_BLEND_FACTOR_ZERO, VK_BLEND_OP_ADD,
+    VK_COLOR_COMPONENT_R_BIT | VK_COLOR_COMPONENT_G_BIT | VK_COLOR_COMPONENT_B_BIT |
+        VK_COLOR_COMPONENT_A_BIT};
+
+const VkPipelineColorBlendAttachmentState m_pSum_BS_render_target_0{
+    true, VK_BLEND_FACTOR_ONE, VK_BLEND_FACTOR_ONE, VK_BLEND_OP_ADD, VK_BLEND_FACTOR_ONE,
+    VK_BLEND_FACTOR_ONE, VK_BLEND_OP_ADD,
+    VK_COLOR_COMPONENT_R_BIT | VK_COLOR_COMPONENT_G_BIT | VK_COLOR_COMPONENT_B_BIT |
+        VK_COLOR_COMPONENT_A_BIT};
+
+// Full screen quad layout structure
+const VkVertexInputBindingDescription layout_quad_bindings[1] = {
+    {0, 32, VK_VERTEX_INPUT_RATE_VERTEX}};
+const VkVertexInputAttributeDescription layout_quad[3] = {
```

Comment:
If it's a fullscreen quad, then call it `fullscreen_quad`. Also, why is there a _tangent_ on a full-screen quad?